chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -2105,63 +2056,31 @@ package experimental {
-    def apply(): Interval = apply(Width(), UnknownBinaryPoint)
+    def apply(): Interval = Interval(range"[?,?]")
 
     /** Create a Interval type with specified width. */
-    def apply(width: Width): Interval = Interval(width, 0.BP)
+    def apply(binaryPoint: BinaryPoint): Interval = {
+      val binaryPointString = binaryPoint match {
+        case KnownBinaryPoint(value) => s"$value"
+        case _ => s""
+      }
+      Interval(range"[?,?].$binaryPointString")
+    }
 
     /** Create a Interval type with specified width. */
-    def apply(width: Width, binaryPoint: BinaryPoint): Interval = {
-      Interval(width, new IntervalRange(firrtlir.UnknownBound, firrtlir.UnknownBound, IntervalRange.getBinaryPoint(binaryPoint)))
-    }
+    def apply(width: Width): Interval = Interval(width, 0.BP)
 
-    // TODO: (chick) Might want to round the range for asInterval as well?
     /** Create a Interval type with specified width. */
-    def apply(width: Width, range: chisel3.internal.firrtl.IntervalRange): Interval = {
-      (range.lower, range.upper, range.binaryPoint) match {
-        case (lx: firrtlir.Bound, ux: firrtlir.Bound, KnownBinaryPoint(bp)) =>
-          // No mechanism to pass open/close to firrtl so need to handle directly -> convert to firrtlir.Closed
-          // TODO: (chick) can we pass open/close to firrtl?
-          val lower = lx match {
-            case firrtlir.Open(x) =>
-              val l = x + BigDecimal(1) / BigDecimal(BigInt(1) << bp)
-              val min = (l * BigDecimal(BigInt(1) << bp)).setScale(0, BigDecimal.RoundingMode.FLOOR) / BigDecimal(BigInt(1) << bp)
-              firrtlir.Closed(min)
-            case firrtlir.Closed(x) =>
-              val l = x
-              val min = (l * BigDecimal(BigInt(1) << bp)).setScale(0, BigDecimal.RoundingMode.FLOOR) / BigDecimal(BigInt(1) << bp)
-              firrtlir.Closed(min)
-            case _ =>
-              lx
-          }
-          val upper = ux match {
-            case firrtlir.Open(x) =>
-              val u = x - BigDecimal(1) / BigDecimal(BigInt(1) << bp)
-              val max = (u * BigDecimal(BigInt(1) << bp)).setScale(0, BigDecimal.RoundingMode.CEILING) / BigDecimal(BigInt(1) << bp)
-              firrtlir.Closed(max)
-            case firrtlir.Closed(x) =>
-              val u = x
-              val max = (u * BigDecimal(BigInt(1) << bp)).setScale(0, BigDecimal.RoundingMode.CEILING) / BigDecimal(BigInt(1) << bp)
-              firrtlir.Closed(max)
-            case _ =>
-              ux
-          }
-          val newRange = chisel3.internal.firrtl.IntervalRange(lower, upper, range.binaryPoint)
-          new Interval(width, newRange)
-        case _ =>
-          new Interval(width, range)
-      }
+    def apply(width: Width, binaryPoint: BinaryPoint): Interval = {
+      Interval(IntervalRange(width, binaryPoint))
     }
 
-    /** Create a Interval with the specified range */
+    /** Create a Interval type with specified width.
```

Comment:
why specified width, not specified range? change back?

Reply:
fixed comments